web/frontend/src/components/do-command/errors.ts

```diff
@@ -0,0 +1,25 @@
+const GRPC_MESSAGE_KEY = 'grpc-message';
+
+export interface GRPCUnimplementedError {
+  metadata: {
+    headersMap: {
+      [GRPC_MESSAGE_KEY]: string[];
+    };
+  };
+}
+
+export const isUnimplementedError = (error: unknown) => {
+  const unimplemented = error as GRPCUnimplementedError;
+  if (
+    typeof unimplemented !== 'object' ||
+    typeof unimplemented.metadata !== 'object' ||
+    typeof unimplemented.metadata.headersMap !== 'object' ||
+    unimplemented.metadata.headersMap[GRPC_MESSAGE_KEY].length === 0
```

Comment:
This whole thing might be easier with `lodash`, which I assume will do these safety checks for us. Is that something we have available?

```ts
const errorMessages = get(error, `metadata.headersMap.${GRPC_MESSAGE_KEY}`, [])
```

Reply:
Good call, will swap that out.